src/frontend/magnify/src/components/rooms/AddGroupToRoomDialog/AddGroupToRoomDialog.tsx

```diff
@@ -0,0 +1,147 @@
+      'Add {numberSelected} {numberSelected, plural, =0 {group} one {group} other {groups}}',
+    description: 'Label for the add groups button',
+  },
+  cancelLabel: {
+    id: 'components.rooms.AddGroupToRoomDialog.cancelLabel',
+    defaultMessage: 'Cancel',
+    description: 'Label for the cancel button',
+  },
+});
+
+const renderGroup = (group: Group | undefined, onRemove: (id: string) => void) => {
+  if (!group) return null;
+  return (
+    <Button
+      key={group.id}
+      onClick={(event: React.MouseEvent) => {
+        event.preventDefault();
+        event.stopPropagation();
+        onRemove(group.id);
+      }}
+      href="#"
+      onFocus={(event: React.FocusEvent) => event.stopPropagation()}
+    >
+      <Box
+        align="center"
+        direction="row"
+        gap="xsmall"
+        pad={{ vertical: 'xsmall', horizontal: 'small' }}
+        margin="xsmall"
+        background="brand"
+        round="xsmall"
+        width="small"
+        height="calc(100% - 10px)"
+      >
+        <Text size="small">
+          {group.name} ({group.members?.length})
+        </Text>
+        <Box round="full" margin={{ left: 'xsmall' }}>
+          <FormClose size="small" style={{ width: '12px', height: '12px' }} />
+        </Box>
+      </Box>
+    </Button>
+  );
+};
+
+export interface AddGroupToRoomDialogProps {
+  open: boolean;
+  onClose: () => void;
+  roomSlug: string;
+}
+
+const AddGroupToRoomDialog = ({ open, onClose, roomSlug }: AddGroupToRoomDialogProps) => {
+  const intl = useIntl();
+  const controller = useController();
+  const { data: groups, isLoading: groupsLoading } = useQuery('groups', controller.getGroups);
+  const { mutate, isLoading } = useMutation(controller.addGroupsToRoom, {
+    onSuccess: () => onClose(),
+  });
+  const [selected, setSelected] = useState<number[]>([]);
+
+  const handleUnselect = (id: string) => {
+    setSelected(selected.filter((groupIndex) => groups?.[groupIndex]?.id !== id));
+  };
+
+  if (!open) return null;
+  return (
+    <Layer
+      id="confirmDelete"
+      position="center"
+      onClickOutside={onClose}
+      onEsc={onClose}
+      role="dialog"
+    >
+      <Box pad="medium" width="large">
+        <Heading level={3} size="small" color="brand">
+          {intl.formatMessage(messages.AddGroupToRoomDialogTitle)}
+        </Heading>
+        {groupsLoading ? (
+          <WaitingRow />
+        ) : (
+          <Select
+            closeOnChange={false}
+            multiple
+            disabled={isLoading}
+            value={
+              <Box wrap direction="row">
+                {selected && selected.length > 0 ? (
+                  selected.map((groupIndex) => renderGroup(groups?.[groupIndex], handleUnselect))
+                ) : (
+                  <Box pad="small">{intl.formatMessage(messages.selectGroupsToAdd)}</Box>
+                )}
+              </Box>
+            }
+            options={(groups || []).map((g) => g.name)}
```

Comment:
```suggestion
            options={(groups || []).map((group) => group.name)}
```
Please don't use such shortened variable names.

Reply:
Fair enough